backend/fms_core/template_importer/importers/library_preparation.py

```diff
@@ -0,0 +1,100 @@
+        try:
+            self.preloaded_data['process_properties'] = \
+                {o.name: {'property_type_obj': o} for o in list(PropertyType.objects.filter(object_id=protocol.id))}
+        except Exception as e:
+            self.base_errors.append(f"Property Type could not be found. {e}")
+
+    def import_template_inner(self):
+        """
+            LIBRARIES SHEET
+        """
+        libraries_sheet = self.sheets['Library']
+        libraries_rows_data = defaultdict(list)
+        for i, row_data in enumerate(libraries_sheet.rows):
+            library = {'library_batch_id': row_data['Library Batch ID'],
+                       'volume_used': float_to_decimal_and_none(row_data['Sample Volume Used (uL)']),
+                       'comment': row_data['Comment'],
+                       'container':
+                           {'barcode': row_data['Library Container Barcode'],
+                            'coordinates': row_data['Library Container Coordinates'],
+                            'name': row_data['Library Container Name'],
+                            'kind': row_data['Library Container Kind'],
+                            'parent_barcode': row_data['Library Parent Container Barcode'],
+                            'parent_coordinates': row_data['Library Container Coordinates']
+                            },
+                       'volume': float_to_decimal_and_none(row_data['Library Volume']),
+                       'index': row_data['Index'],
+                      }
+
+            sample_kwargs = dict(
+                barcode=row_data['Sample Container Barcode'],
+                coordinates=row_data['Sample Container Coordinates'],
+                volume_used=library['volume_used']
+            )
+
+            (result, library['sample_obj']) = self.handle_row(
+                row_handler_class=SampleRowHandler,
+                sheet=libraries_sheet,
+                row_i=i,
+                **sample_kwargs,
+            )
+
+            libraries_rows_data[library['library_batch_id']].append(library)
+
+        """
+            LIBRARY BATCH SHEET
+        """
+        library_batch_sheet = self.sheets['Library Batch']
+
+        # Iterate through libraries rows
+        for row_id, row in enumerate(library_batch_sheet.rows):
+            library_batch_dict = {}
+            process_properties = self.preloaded_data['process_properties']
+            for i, (key, val) in enumerate(row.items()):
+                if i < self.properties_starting_index:
+                    library_batch_dict[key] = row[key]
+                else:
+                    process_properties[key]['value'] = val
+
+            library_preparation_kwargs = dict(
+                # Library attributes data dictionary and related objects
+                library_type={'name' : library_batch_dict['Experiment Type']},
+                library_size=process_properties['Shearing Size (bp)']['value'],
```

Comment:
We'll have to check this with the lab. We'll have to clarify with the lab if library_size = shearing_size and in that case what would be called the size of the library including the index + adapters/primers.